### PR TITLE
hazelcast update _helpers to add template render function used inside…

### PR DESCRIPTION
**Pull request by bbaudelet**
Add Function on _helper to take some values as template and used it to read as template :

- .Values.podLabels
- .Values.initContainers
- .Values.sidecarContainers

Add extraVolumes and extraVolumeMounts parameters to allow user to mount extra volume to pod.

Add ServiceMinitor to use with prometheus operator to scrape metrics from exporter.

### CODE REVIEW

File: stable/hazelcast/templates/statefulset.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ spec:
         app.kubernetes.io/managed-by: "{{ .Release.Service }}"
         role: hazelcast
         {{- if .Values.podLabels }}
-{{ toYaml .Values.podLabels | indent 8 }}
+{{ include "hazelcast.tplvalues.render" (dict "value" .Values.podLabels "context" $) | indent 8 }}
```
Comment (itscaro) [inline stable/hazelcast/templates/statefulset.yaml:28]:
```suggestion
{{ include "hazelcast.tplvalues.render" (dict "value" .Values.podLabels "context" $) | indent 8 }}
```

File: stable/hazelcast/templates/statefulset.yaml [hunk 2]
```diff
@@ -66,9 +66,9 @@ spec:
       {{- end }}
 {{- if .Values.initContainers }}
       initContainers:
-        {{- toYaml .Values.initContainers | nindent 8 }}
+      {{ include "hazelcast.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
 {{- end }}
-      containers:
+      containers:
```
Comment (itscaro) [inline stable/hazelcast/templates/statefulset.yaml:71]:
```suggestion
      containers:
```

File: stable/hazelcast/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      scrapeTimeout: {{ .Values.metrics.serviceMonitor.scrapeTimeout }}
+      {{- end }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels: 
+      app.kubernetes.io/name: {{ template "hazelcast.name" . }}
+      app.kubernetes.io/instance: "{{ .Release.Name }}"
+{{- end }}
```
Comment (itscaro) [inline stable/hazelcast/templates/servicemonitor.yaml:32]:
```suggestion
{{- end }}

```

File: stable/hazelcast/values.yaml [hunk 4]
```diff
@@ -201,10 +201,16 @@ metrics:
     type: ClusterIP
     port: 8080
     portName: metrics
+    labels: {}
     annotations:
       prometheus.io/scrape: "true"
       prometheus.io/path: "/metrics"
       prometheus.io/port: "8080"
+  serviceMonitor:
+      port: metrics
```
Comment (itscaro) [inline stable/hazelcast/values.yaml:210]: Use `{{ .Values.metrics.service.portName }}` instead?

**Comment by hakanmemisoglu** [comment]
@bbaudelet 
Can you sign the CLA?

**Comment by mtyazici** [comment]
Hi, @bbaudelet we have added ServiceMonitor support for Prometheus Operator. Could you sign the CLA, change the Chart versions, do the same changes for enterprise and remove the ServiceMonitor support from the PR?

**Pull request closed without merging**